.github/workflows/nightly.yml

```diff
@@ -0,0 +1,88 @@
+name: 🌒 Nightly Release
+
+on:
+  schedule:
+    - cron: "0 7 * * *" # every day at 12AM PST
+
+jobs:
+  nightly:
+    name: 🌒 Nightly Release
+    if: github.repository == 'remix-run/remix'
+    runs-on: ubuntu-latest
+    outputs:
+      NEXT_VERSION: ${{ steps.version.outputs.NEXT_VERSION }}
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: dev
+          token: ${{ secrets.FORMAT_PAT }}
+          fetch-depth: 0
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: "yarn"
+
+      - name: 📥 Install deps
+        # even though this is called "npm-install" it does use yarn to install
+        # because we have a yarn.lock and caches efficiently.
+        uses: bahmutov/npm-install@v1
+
+      - name: 🏗 Build
+        run: yarn build
+
+      - name: ⤴️ Update Version
+        id: version
+        run: |
+          SHA=$(git rev-parse HEAD)
+          SHORT_SHA=${SHA::7}
+          NEXT_VERSION=0.0.0-nightly-${SHORT_SHA}
```

Comment:
i'm in between using the date instead of the commit sha, i think some folks use both?